sdk/tables/Azure.Data.Tables/src/TableEntity.cs

```diff
@@ -204,6 +204,7 @@ private protected void SetValue(string key, object value)
         /// <exception cref="InvalidOperationException">Value associated with given <paramref name="key"/> is not of type <paramref name="type"/>.</exception>
         private protected object GetValue(string key, Type type = null)
```

Comment:
To the discussion about how to handle keys not present in the dictionary, I'm leaning towards throwing `KeyNotFoundException` to distinguish between that and the value being present but null. The storage service [doesn't distinguish between properties being null and not present](https://docs.microsoft.com/en-us/rest/api/storageservices/merge-entity#remarks) although we should validate this is also true for Cosmos endpoints. Given that behavior, being clear when it doesn't exist might communicate a more consistent mental model of how nulls will be treated. However, one could argue that to be truly consistent with service behavior we should treat null as a `KeyNotFoundException`, but that would violate expectations about the behavior of an `IDictionary`.

I'm curious if @tg-msft has any thoughts here.

Reply:
[We throw in Search, but the service semantics are different.](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/src/SearchDocument/SearchDocument.Core.cs#L55)  I think returning `null` is the safest choice here per the linked documentation.  Anyone who wants to explicitly check existence could use `TryGetValue` or check the `Keys` property.  Everyone else will just be glad they're not getting exceptions for all their optional properties.  This would be worth discussing during any additional Arch Board reviews though.